core/src/test/java/org/elasticsearch/snapshots/SharedClusterSnapshotRestoreIT.java

```diff
@@ -2490,4 +2493,80 @@ public void testGetSnapshotsRequest() throws Exception {
+
+        logger.info("--> creating repository");
+        PutRepositoryResponse putRepositoryResponse =
+            client.admin().cluster().preparePutRepository(repo).setType("fs").setSettings(Settings.builder()
+                .put("location", randomRepoPath())
+                // very slow snapshotting, so the node has time to leave the cluster and rejoin while still snapshotting
+                .put("max_snapshot_bytes_per_sec", "100b")
+            ).get();
+        assertTrue(putRepositoryResponse.isAcknowledged());
+
+        logger.info("--> snapshot");
+        client.admin().cluster().prepareCreateSnapshot(repo, snapshot)
+            .setWaitForCompletion(false)
+            .execute();
+
+        logger.info("--> waiting for snapshot to be in progress on all nodes");
+        assertBusy(() -> {
+            for (String node : internalCluster().nodesInclude(index)) {
+                final Client nodeClient = client(node);
+                SnapshotsInProgress snapshotsInProgress = nodeClient.admin().cluster().prepareState().get()
+                                                              .getState().custom(SnapshotsInProgress.TYPE);
+                assertNotNull(snapshotsInProgress);
+                assertEquals(1, snapshotsInProgress.entries().size());
+                assertEquals(snapshot, snapshotsInProgress.entries().get(0).snapshot().getSnapshotId().getName());
+                ImmutableOpenMap<ShardId, SnapshotsInProgress.ShardSnapshotStatus> shards = snapshotsInProgress.entries().get(0).shards();
+                assertEquals(numPrimaries, shards.size());
+                for (ObjectObjectCursor<ShardId, ShardSnapshotStatus> cursor : shards) {
+                    assertEquals(SnapshotsInProgress.State.INIT, cursor.value.state());
+                }
+            }
+        }, 10, TimeUnit.SECONDS);
+
+        // Pick a node with a primary shard and remove the shard from the node
+        ClusterState clusterState = internalCluster().clusterService(internalCluster().getMasterName()).state();
+        IndexRoutingTable indexRoutingTable = clusterState.getRoutingTable().index(index);
+        String nodeWithPrimary = clusterState.nodes().get(indexRoutingTable.shard(0).primaryShard().currentNodeId()).getName();
+        assertNotNull("should be at least one node with a primary shard", nodeWithPrimary);
+        IndicesService indicesService = internalCluster().getInstance(IndicesService.class, nodeWithPrimary);
+        indicesService.deleteIndex(resolveIndex(index), "trigger shard removal");
+
+        // make sure snapshot is aborted
+        assertBusy(() -> {
+            SnapshotsInProgress snapshotsInProgress =
+                client.admin().cluster().prepareState().get().getState().custom(SnapshotsInProgress.TYPE);
+            if (snapshotsInProgress != null && snapshotsInProgress.entries().size() > 0) {
+                assertEquals(State.SUCCESS, snapshotsInProgress.entries().get(0).state());
```

Comment:
Without the change here, the snapshot forever stalls and the test times out, because the snapshot was never aborted.  This asserts that we abort the snapshot, bringing the snapshotting to a successful conclusion.